sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -57,6 +59,9 @@ export class BatchingReceiver extends MessageReceiver {
     // Clears the token renewal timer. Closes the link and its session if they are open.
     await this._closeLink(this._receiver);
     this.detachedError = receiverError;
+    if (this.isReceivingMessages && typeof this._finalActionHandler === "function") {
```

Comment:
Silly question but isn't it enough to just check that (this._finalActionHandler != null)? It's typed to always be a function, is it not?

Reply:
That would be fine. It's just a habit of mine from pre-TS days to ensure that something is a function before invoking it if it might otherwise not exist. I'd still want to do this if `_finalActionHandler` was provided by a user of the SDK (or do some validation elsewhere so we could throw an error).